### ISSUE TITLE
Stop/resume fetching process

### ISSUE THREAD

**Issue by fturco** [libertysoft3/reddit-html-archiver#12]
Is it safe to stop and then resume the fetching process with ctrl+c?
Can that lead to data corruption or inconsistency?
Does `fetch_links.py` properly save data before exiting?

**Comment by libertysoft3** [comment]
Yes, it is safe to quit and re-start, although it is not well tested. I would personally wait until the script outputs "got %s links, wrote %s and %s comments", which happens every 10 links/posts. Waiting for this will avoid quitting the script while write_links() is running.

Since data is written every 10 links/posts, by quitting the most data you will lose is 10 links.

**Comment by libertysoft3** [comment]
If you know how to improve this I'm all ears. I suck at python.

**Comment by libertysoft3** [comment]
Looking at this in more detail, it is safe to re-run a data fetch, but it's not smart at all. It will re-download all of the data. So for now, for sanity/efficiency, you have to check to see the last date that you downloaded data for, and start your next run on that same date.
